lib/solana.cpp

```diff
@@ -501,6 +503,27 @@ Connection::getSignatureStatus(const std::string &signature,
   return {res.context, res.value[0]};
 }
 
+bool Connection::confirmTransaction(std::string transactionSignature,
+                                    Commitment confirmLevel,
+                                    uint16_t retries) const {
+  const auto timeoutBlockheight =
+      getLatestBlockhash(confirmLevel).lastValidBlockHeight +
+      solana::MAXIMUM_NUMBER_OF_BLOCKS_FOR_TRANSACTION;
+  auto currentBlockheight = getBlockHeight(confirmLevel);
+  while (retries > 0) {
+    const auto res = getSignatureStatus(transactionSignature, true).value;
```

Comment:
Everything looks ok just a minor change remaining: 
`auto currentBlockheight = getBlockHeight(confirmLevel);`
Could be done once in while loop just before the comparason with timeoutBlockheight. 
So we have one line instead of 2.

Reply:
Oh yeah must have gotten left over 
Moved it into the while loop